langchain-core/src/prompts/chat.ts

```diff
@@ -307,6 +310,274 @@ export class ChatMessagePromptTemplate<
   }
 }
 
+interface _TextTemplateParam {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  text?: string | Record<string, any>;
+}
+
+interface _ImageTemplateParam {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  image_url?: string | Record<string, any>;
+}
+
+class _StringImageMessagePromptTemplate<
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  RunInput extends InputValues = any,
+  RunOutput extends BaseMessage[] = BaseMessage[]
+> extends BaseMessagePromptTemplate<RunInput, RunOutput> {
+  lc_namespace = ["langchain_core", "prompts", "chat"];
+
+  lc_serializable = true;
+
+  inputVariables: Array<Extract<keyof RunInput, string>> = [];
+
+  additionalOptions: Record<string, unknown> = {};
+
+  prompt:
+    | BaseStringPromptTemplate<
+        InputValues<Extract<keyof RunInput, string>>,
+        string
+      >
+    | Array<
+        | BaseStringPromptTemplate<
+            InputValues<Extract<keyof RunInput, string>>,
+            string
+          >
+        | ImagePromptTemplate<
+            InputValues<Extract<keyof RunInput, string>>,
+            string
+          >
+        | MessageStringPromptTemplateFields<
+            InputValues<Extract<keyof RunInput, string>>
+          >
+      >;
+
+  _messageClass:
+    | typeof HumanMessage
+    | typeof AIMessage
+    | typeof SystemMessage
+    | undefined;
+
+  // ChatMessage contains role field, others don't.
+  // Because of this, we have a separate class property for ChatMessage.
+  _chatMessageClass: typeof ChatMessage | undefined;
```

Comment:
Where did this `_` pattern come from for instance vars?

Why don't we just make it `protected`?

Reply:
Can do, was copying python but `protected` works too